### PR TITLE
Add MPS validation tool

### PR DESCRIPTION
**Pull request by prigoll**
New tool to centrally validate the results of Millepede Alignment Fit
Runs standalone and integrated in the MPS workflow
The tool was presented and discussed at the Millepede Alignment Meeting 2016-07-26 https://indico.cern.ch/event/547579/

### CODE REVIEW

**Comment by mmusich** [comment]
please test

File: Alignment/MillePedeAlignmentAlgorithm/python/mpsvalidate/htmlCreator.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+        for plot in [x for x in config.outputList if x.plottype == "monitor"]:
+            out += "<h3>{0}</h3>\n".format(plot.name)
+            out += "<a href='plots/pdf/{0}.pdf'><img src='plots/png/{0}.png'></a>\n".format(
+                plot.filename)
+
+    data = data.substitute(message=config.message, out=out)
+
+    with open(os.path.join(config.outputPath, outputFile), "w") as output:
+        output.write(data)
+        output.close()
```
Comment (ghellwig) [inline Alignment/MillePedeAlignmentAlgorithm/python/mpsvalidate/htmlCreator.py]: @prigoll you are using a 'with' construct which automatically closes the file, i.e. this line is redundant

File: Alignment/MillePedeAlignmentAlgorithm/python/mpsvalidate/html_template.html [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <title>Project %%message</title>
+</head>
+<body>
+%%out
+</body>
+</html>
```
Comment (ghellwig) [inline Alignment/MillePedeAlignmentAlgorithm/python/mpsvalidate/html_template.html]: @prigoll no newline at the end of the file

File: Alignment/MillePedeAlignmentAlgorithm/python/mpsvalidate/iniparser.py [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+        if (parser.read(path) == []):
+            logger.error("Could not open ini-file: {0}".format(path))
+
+        # buffer object
+        configBuffer = ConfigData()
+
+        # collect data and process it
+        try:
+            configBuffer.jobNumber = int(parser.get("GENERAL", "job"))
+        except:
```
Comment (ghellwig) [inline Alignment/MillePedeAlignmentAlgorithm/python/mpsvalidate/iniparser.py]: @prigoll here and below you are catching all exceptions w/o specifying what exceptions you expect here, i.e. you will also silently catch exceptions that you do not expect;
please be explicit here

File: Alignment/MillePedeAlignmentAlgorithm/python/mpsvalidate/iniparser.py [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+        if (args.message != ""):
+            self.message = args.message
+
+        # if path is given put the output in the current directory
+        if (args.jobdatapath):
+            self.outputPath = "validation_output"
+        
+        if (args.outputpath):
+            self.outputPath = args.outputpath
+
```
Comment (ghellwig) [inline Alignment/MillePedeAlignmentAlgorithm/python/mpsvalidate/iniparser.py]: @prigoll no newline at the end of the file

File: Alignment/MillePedeAlignmentAlgorithm/scripts/mps_runPede_rfcp_template.sh [hunk 5]
```diff
@@ -182,10 +182,12 @@ gzip -f millepede.eve
...
-for tag in $(sqlite3 alignments_MP.db  "SELECT NAME FROM TAG;")
-do
-    conddb --db alignments_MP.db list ${tag}
-done
+conddb --db alignments_MP.db list Alignments
+conddb --db alignments_MP.db list Deformations
+conddb --db alignments_MP.db list SiStripLorentzAngle_deco
+conddb --db alignments_MP.db list SiStripLorentzAngle_peak
+conddb --db alignments_MP.db list SiPixelLorentzAngle
+conddb --db alignments_MP.db list SiStripBackPlaneCorrection
```
Comment (ghellwig) [inline Alignment/MillePedeAlignmentAlgorithm/scripts/mps_runPede_rfcp_template.sh]: @prigoll you are reintroducing old code here; the for loop is the more general and cleaner version of the code you are reintroducing here

File: Alignment/MillePedeAlignmentAlgorithm/scripts/mps_validate.py [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+import os
+import shutil
+import sys
+
+from ROOT import PyConfig
+PyConfig.IgnoreCommandLineOptions = True
+
+from ROOT import (TH1F, TCanvas, TFile, TImage, TPaveLabel, TPaveText, TTree,
+                  gROOT, gStyle)
+PyConfig.IgnoreCommandLineOptions = True
```
Comment (ghellwig) [inline Alignment/MillePedeAlignmentAlgorithm/scripts/mps_validate.py]: @prigoll duplication of line 17

File: Alignment/MillePedeAlignmentAlgorithm/test/trackerTree_cfg.py [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+# Input source
+process.source = cms.Source("EmptySource")
+
+# Number of events
+process.maxEvents = cms.untracked.PSet( input = cms.untracked.int32(1) )
+
+# Geometry
+#process.load("Configuration.StandardSequences.Geometry_cff")
+#process.load("Configuration.Geometry.GeometryIdeal_cff")
```
Comment (ghellwig) [inline Alignment/MillePedeAlignmentAlgorithm/test/trackerTree_cfg.py]: @prigoll please remove the commented code

**Comment by mmusich** [comment]
@prigoll if you comply with Gregor's comment by today at 5 chances are we can ask this to be integrated at the ORP meeting

**Comment by mmusich** [comment]
please test

**Comment by ghellwig** [comment]
please test

**Comment by ghellwig** [comment]
+1
provided jenkins agrees

**Comment by mschrode** [comment]
@mmusich please let us know if you need any further information for the ORP meeting. Tracker alignment conveners support this PR :)

**Comment by davidlange6** [comment]
+1

**Pull request merged**